includes/core/classes/class-event-feed.php

```diff
@@ -0,0 +1,321 @@
+		// Only run on the main query and if it's a feed.
+		if ( ! $query->is_main_query() || ! is_feed() ) {
+			return;
+		}
+
+		// Check if this is an events feed request.
+		if ( isset( $_SERVER['REQUEST_URI'] ) ) {
+			$request_uri = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+			// Check if this is the events feed URL.
+			if ( str_contains( $request_uri, '/' . $this->rewrite_slug . '/' . $GLOBALS['wp_rewrite']->feed_base ) ) {
+				// Set the post type and let Event_Query handle the rest.
+				$query->set( 'post_type', Event::POST_TYPE );
+				$query->set( 'gatherpress_events_query', 'upcoming' );
+			}
+		}
+	}
+
+	/**
+	 * Get formatted event datetime information for feeds.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param Event $event The event object.
+	 * @return array Array of event information strings.
+	 */
+	private function get_event_datetime_info( Event $event ): array {
+		$event_info       = array();
+		$display_datetime = $event->get_display_datetime();
+
+		if ( ! empty( $display_datetime ) && '—' !== $display_datetime ) {
+			$event_info[] = sprintf(
+				/* translators: %s: Formatted date and time */
+				__( 'Date: %s', 'gatherpress' ),
+				$display_datetime
+			);
+		}
+
+		return $event_info;
+	}
+
+	/**
+	 * Get default event excerpt customization.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $excerpt The current excerpt.
+	 * @return string The customized excerpt.
+	 */
+	public function get_default_event_excerpt( string $excerpt ): string {
+		global $post;
+
+		// Only apply to events.
+		if ( Event::POST_TYPE !== get_post_type( $post ) ) {
+			return $excerpt;
+		}
+
+		$event = new Event( get_the_ID() );
+		$venue = $event->get_venue_information();
+
+		$event_info = $this->get_event_datetime_info( $event );
+
+		// Add venue information.
+		if ( $venue && ! empty( $venue['name'] ) ) {
+			$event_info[] = sprintf(
+				/* translators: %s: Venue name */
+				__( 'Venue: %s', 'gatherpress' ),
+				$venue['name']
+			);
+		}
+
+		// Build the customized excerpt.
+		$custom_excerpt = '';
+
+		if ( ! empty( $event_info ) ) {
+			$custom_excerpt .= '<p><strong>' . implode( ' | ', $event_info ) . '</strong></p>';
+		}
+
+		// Add the original excerpt if it exists and is different from the content.
+		if ( ! empty( $excerpt ) && get_the_content() !== $excerpt ) {
+			$clean_excerpt = wp_strip_all_tags( $excerpt );
+			$clean_excerpt = preg_replace( '/\s+/', ' ', $clean_excerpt ); // Normalize whitespace.
+			$clean_excerpt = trim( $clean_excerpt );
+
+			if ( ! empty( $clean_excerpt ) ) {
+				$custom_excerpt .= '<p>' . $clean_excerpt . '</p>';
+			}
+		}
+
+		return $custom_excerpt;
+	}
+
+	/**
+	 * Get default event content customization.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $content The current content.
+	 * @return string The customized content.
+	 */
+	public function get_default_event_content( string $content ): string {
+		global $post;
```

Comment:
_**Same as above:**_ I would remove the `global $post `and rely on `get_post_type()` falling back to the global internaly.